src/main/java/com/planner/ui/CLI.java

```
@@ -86,7 +88,12 @@ private void exeCmd(String[] tokens) throws IOException {
                 break;
             case "event":
                 if (tokens.length > 1) {
-                    // TODO
+                    Parser.EventInfo eventInfo = Parser.parseEvent(tokens);
+
+                    Time.TimeStamp timeStamp = new Time.TimeStamp(eventInfo.getTimestamp()[0], eventInfo.getTimestamp()[1]);
+
+                    Event event = sm.addEvent(eventInfo.getName(), eventInfo.getCardId(), timeStamp, eventInfo.isRecurring(), eventInfo.getDates());
```

If `eventInfo` is **NOT** recurring, set `List<Calendar> dates` to null **IF**:

- `eventInfo.getDates()` is null OR `eventInfo.getDates()` has only 1 date

Note1: Individual Events are allowed to have **up to** 1 date. If no date is provided, it's assumed to be the current day (which is already handled when parsing the TimeStamp info). If a date is provided, then the TimeStamp adopts that date.

Note2: While the parser doesn't require a date for recurring Events, ScheduleManager does. And, to be frank, I think this is a **good** route since it forces the user to be clear about what they want recurring. Not providing a date with a recurring event like the following is unclear and error prone:

- `event "study" +C2 @ 11-3` (assume that today is Monday, for context)

Does this mean recur everyday or only on Mondays? This is why recurring Events should require dates.

Question: Should we throw the error in Parser or in ScheduleManager if no date is provided for a recurring Event?

---

Yes, I think we should to avoid confusion.

---

Totally agree, will create a new PR for that.